packages/block-library/src/social-link/index.php

```diff
@@ -57,13 +53,21 @@ function render_block_core_social_link( $attributes, $content, $block ) {
 	);
 
 	$link  = '<li ' . $wrapper_attributes . '>';
-	$link .= '<a href="' . esc_url( $url ) . '" ' . $rel_target_attributes . ' class="wp-block-social-link-anchor">';
+	$link .= '<a href="' . esc_url( $url ) . '" class="wp-block-social-link-anchor">';
 	$link .= $icon;
 	$link .= '<span class="wp-block-social-link-label' . ( $show_labels ? '' : ' screen-reader-text' ) . '">';
 	$link .= esc_html( $label );
 	$link .= '</span></a></li>';
 
-	return $link;
+	$w = new WP_HTML_Tag_Processor($link);
+	$w->next_tag('a');
+	if ( $open_in_new_tab ) {
+		$w->set_attribute('rel', esc_attr( $rel ) . ' noopener nofollow');
+		$w->set_attribute('target', '_blank');
+	} else if ( $rel !== '' ) {
+		$w->set_attribute('rel', esc_attr( $rel) );
+	}
+	return $w;
```

Comment:
nice use of the tag processor!

Reply:
It seemed like a clearer solution than a bunch of string manipulation and worked like a charm!

Reply:
Nit: While I also like how clean the "tag processor" looks here, but I don't think it is needed. We're building whole block output in the render callback; in cases like this, I prefer PHP as a templating engine.

Reply:
@Mamaduka I'm confused on what you are saying here. I guess I can see that we're constructing the link HTML entirely, we're not modifying content that was given to us, in which case we could change this whole thing to something like…

```php
$anchor = $open_in_new_tab
	? '<a href="' . esc_url( $url ) . '" rel="' . esc_attr( $rel ) . ' noopener nofollow" target="_blank" class="wp-block-social-link">'
	: '<a href="' . esc_url( $url ) . '" class="wp-block-social-link">`;

…

$link .= $anchor;
```

that makes sense, though it does highlight a use-case for the tag processor that maybe wasn't the primary goal when creating it, which is making tag text construction more obvious. personally I often find these template pieces noisy and hard to follow, especially since they tend to make for very long lines with intertwining PHP and strings.

```php
$anchor = new WP_HTML_Tag_Processor( "<a>" );
$anchor->next_tag( 'a' );
$anchor->set_attribute( 'href', $url );
$anchor->set_attribute( 'class', 'wp-block-social-link-anchor' );
if ( $open_in_new_tab ) {
	$anchor->set_attribute( 'rel', "{$rel} no opener nofollow" );
	$anchor->set_attribute( 'target', '_blank' );
} else {
	$anchor->set_attribute( 'rel', $rel );
}
$link .= $anchor;
```

now I think maybe there could be a good helper for this 🤔 

```php
static function make_tag_opener( $tag_name, $attributes = [] ) {
	$w = new WP_HTML_Tag_Walker( "<{$tag_name}>" );
	$w->next_tag( $tag_name );

	foreach ( $attributes as $name => $value ) {
		$w->set_attribute( $name, $value );
	}

	return $w;
} 
```

this code here would then turn into something even less noisy, and at the same time, properly encoded.

```php
$anchor = WP_HTML_Tag_Processor::make_tag_opener( 'a', array(
	'href'  => $url,
	'class' => 'wp-block-social-anchor',
	'rel'   => $rel
) );

if ( $open_in_new_tab ) {
	$anchor->set_attribute( 'rel', "{$rel} no opener nofollow" );
	$anchor->set_attribute( 'target', '_blank' );
}

…

$link .= $anchor;
```

none of this is to contradict your input; just taking the opportunity to muse on how this could evolve. this is a very DOM-like interface with its benefits as well as its drawbacks.

cc: @adamziel